client.go

```diff
@@ -179,6 +177,7 @@ func (c *Client) exchange(m *Msg, a string) (r *Msg, rtt time.Duration, err erro
 	co.TsigSecret = c.TsigSecret
 	// write with the appropriate write timeout
 	co.SetWriteDeadline(time.Now().Add(c.getTimeoutForRequest(c.writeTimeout())))
+	t := time.Now()
```

Comment:
If you moved this up a line, we could avoid calling `time.Now` twice. It’s a small thing, but it does come with overhead.

Reply:
Fixed.